src/lib/algolia/algoliaSearchProvider.ts

```diff
@@ -16,78 +17,159 @@ export interface AlgoliaSearchProviderOptions {
   indexNamePrefix?: string;
 }
 
-// @todo implement batch operations
 export class AlgoliaSearchProvider implements SearchProvider {
   #algolia: SearchClient;
-
-  indexNamePrefix?: string;
+  #indexNamePrefix?: string | undefined;
 
   constructor({ appId, apiKey, indexNamePrefix }: AlgoliaSearchProviderOptions) {
     this.#algolia = Algoliasearch(appId, apiKey);
-    this.indexNamePrefix = indexNamePrefix;
+    this.#indexNamePrefix = indexNamePrefix;
+  }
+
+  private async saveGroupedByIndex(groupedByIndex: GroupedByIndex) {
+    return Promise.all(
+      Object.entries(groupedByIndex).map(([indexName, objects]) => {
+        const index = this.#algolia.initIndex(indexName);
+        return index.saveObjects(objects);
```

Comment:
The most important change is here. We first group variants by indexName and then save many of them at once using `saveObjects` Algolia method.